File: Lesson - 2/2.js
function solution(A) {
    A.sort();
    for (let i = 0; i + 1 < A.length; i += 2) {
        if (A[i] != A[i + 1]) {
            return A[i];
        }
    }
    return A[A.length - 1];
}

File: Lesson - 7/2.java
import java.lang.*;
import java.util.*;

class Solution {
    public int solution(String S) {
        if (S.length() == 0) {
            return 1;
        }
        
        int openingBraces = 0;
        for (char letter : S.toCharArray()) {
            if (letter == '(') {
                openingBraces++;
            }
            else {
                --openingBraces;
                if (openingBraces < 0) {
                    return 0;
                }
            }
        }
        return openingBraces == 0 ? 1 : 0;
    }
}